pkg/devfile/generator/generators.go

```diff
@@ -168,25 +169,28 @@ func GetInitContainers(devfileObj parser.DevfileObj) ([]corev1.Container, error)
 	return initContainers, nil
 }
 
-// DeploymentParams is a struct that contains the required data to create a deployment object
-type DeploymentParams struct {
-	TypeMeta          metav1.TypeMeta
-	ObjectMeta        metav1.ObjectMeta
-	InitContainers    []corev1.Container
-	Containers        []corev1.Container
-	Volumes           []corev1.Volume
-	PodSelectorLabels map[string]string
-	Replicas          *int32
+// PodTemplateParams is a struct that contains the required data to create a podtemplatespec object
+type PodTemplateParams struct {
+	ObjectMeta metav1.ObjectMeta
+	// PodSecurityAdmissionPolicy is the policy to be respected by the created pod
+	// The pod will be patched, if necessary, to respect the policies
+	PodSecurityAdmissionPolicy psaapi.Policy
 }
 
-// GetDeployment gets a deployment object
-func GetDeployment(devfileObj parser.DevfileObj, deployParams DeploymentParams) (*appsv1.Deployment, error) {
+func GetPodTemplateSpec(devfileObj parser.DevfileObj, podTemplateParams PodTemplateParams) (*corev1.PodTemplateSpec, error) {
+	containers, err := GetContainers(devfileObj, common.DevfileOptions{})
+	if err != nil {
+		return nil, err
+	}
+	initContainers, err := GetInitContainers(devfileObj)
+	if err != nil {
+		return nil, err
+	}
 
 	podTemplateSpecParams := podTemplateSpecParams{
-		ObjectMeta:     deployParams.ObjectMeta,
-		InitContainers: deployParams.InitContainers,
-		Containers:     deployParams.Containers,
-		Volumes:        deployParams.Volumes,
+		ObjectMeta:     podTemplateParams.ObjectMeta,
+		InitContainers: initContainers,
+		Containers:     containers,
```

Comment:
I think we should also set the `Volumes` field here, no? Perhaps by exposing it in the `PodTemplateParams` struct..
Maybe I missed it, but I couldn't see how Volumes could be defined easily.

Reply:
On Odo, they are set afterwards, on the returned podtemplatespec. That seems easier, as we also want to set the volumemounts in the containers at the same time.